numpy/lib/_function_base_impl.py

```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-
-    .. math::
-        i + g = q * ( n - alpha - beta + 1 ) + alpha
-
-    The different methods then work as follows
-
-    inverted_cdf:
-        method 1 of H&F [1]_.
-        This method gives discontinuous results:
-
-        * if g > 0 ; then take j
-        * if g = 0 ; then take i
-
-    averaged_inverted_cdf:
-        method 2 of H&F [1]_.
-        This method gives discontinuous results:
-
-        * if g > 0 ; then take j
-        * if g = 0 ; then average between bounds
+    By default (``method='linear'``), this is done by interpolating between
+    adjacent elements in ``y``, a sorted copy of `a`::
 
-    closest_observation:
-        method 3 of H&F [1]_.
-        This method gives discontinuous results:
+        (1-g)*y[j] + g*y[j+1]
 
-        * if g > 0 ; then take j
-        * if g = 0 and index is odd ; then take j
-        * if g = 0 and index is even ; then take i
+    where the index ``j`` and coefficient ``g`` are the integral and
+    fractional components of ``q * (n-1)``, and ``n`` is the number of
+    elements in the sample.
 
-    interpolated_inverted_cdf:
-        method 4 of H&F [1]_.
-        This method gives continuous results using:
+    This a special case of Equation 1 of H&F [1]_. More generally,
 
-        * alpha = 0
-        * beta = 1
+    - ``j = (q*n + m - 1) // 1``, and
+    - ``g = (q*n + m - 1) % 1``,
 
-    hazen:
-        method 5 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 1/2
-        * beta = 1/2
-
-    weibull:
-        method 6 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 0
-        * beta = 0
-
-    linear:
-        method 7 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 1
-        * beta = 1
-
-    median_unbiased:
-        method 8 of H&F [1]_.
-        This method is probably the best method if the sample
-        distribution function is unknown (see reference).
-        This method gives continuous results using:
-
-        * alpha = 1/3
-        * beta = 1/3
-
-    normal_unbiased:
-        method 9 of H&F [1]_.
-        This method is probably the best method if the sample
-        distribution function is known to be normal.
-        This method gives continuous results using:
-
-        * alpha = 3/8
-        * beta = 3/8
-
-    lower:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` as the interpolation point.
-
-    higher:
-        NumPy method kept for backwards compatibility.
-        Takes ``j`` as the interpolation point.
-
-    nearest:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` or ``j``, whichever is nearest.
+    where ``m`` is defined differently for each value of the ``method``
+    parameter.
 
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    =============================== =============== ===============
```

Comment:
This is another thing for the reviewing maintainer to consider. I would agree if the relationship between all the methods were simpler. Unfortunately, methods 1-3 use three different definitions of `g`, which breaks the nice interpretation of `j` and `g` being the "integer and fractional components" of the same quantity. We could add a column to the table, but I left them out intentionally since these methods are quite different.

Reply:
Again, the compromise would be to mention this somehow, as a note to the reader that this table is not complete. Maybe something like move one sentence from a couple paragraphs down and write:

```
For the continuous estimators defined in [1]_, the table below shows how ``m`` is computed as a function of probability `q`:
```

Reply:
There are only three sentences that separate this table of continuous estimators from the discontinuous estimators below. 

The first is a minor footnote about clipping the indices. If we cannot make this an actual footnote, the best place for it is immediately after the table. (Before the table would interrupt the explanation, and a Note admonition would only increase the separation between the table and list.)

The next two explain that the table includes only the continuous estimators and introduces the discontinuous estimators from that same paper. I've modified it slightly (below) to emphasize that the table does not include all the estimators from that paper.